maintenance_plan/models/maintenance.py

```diff
@@ -0,0 +1,219 @@
+                # We use 2 times the period to avoid creation too closed
+                # request from a manually one created
+                if date_gap > timedelta(0) and date_gap > period_timedelta * 2:
+                    next_date = fields.Date.to_string(
+                        today_date + period_timedelta)
+            elif last_maintenance_done:
+                next_date = fields.Date.from_string(
+                    last_maintenance_done.close_date) + period_timedelta
+                # If when we add the period to the last maintenance done and
+                # we still in past, we plan it for today
+                if next_date < today_date:
+                    next_date = date_now
+            else:
+                next_date = fields.Date.to_string(
+                    today_date + period_timedelta)
+
+            plan.next_maintenance_date = next_date
+
+    @api.multi
+    def unlink(self):
+        """ Restrict deletion of maintenance plan should there be maintenance
+            requests of this kind which are not done for its equipment """
+        for plan in self:
+            request = plan.equipment_id.mapped('maintenance_ids').filtered(
+                lambda r: (
+                    r.maintenance_kind_id == plan.maintenance_kind_id and
+                    not r.stage_id.done and
+                    r.maintenance_type == 'preventive')
+            )
+            if request:
+                raise UserError(_('The maintenance plan %s of equipment %s '
+                                  'has generated a request which is not done '
+                                  'yet. You should either set the request as '
+                                  'done, remove its maintenance kind or '
+                                  'delete it first.') % (
+                    plan.maintenance_kind_id.name, plan.equipment_id.name))
+        super(MaintenancePlan, self).unlink()
+
+    _sql_constraints = [
+        ('equipment_kind_uniq', 'unique (equipment_id, maintenance_kind_id)',
+         "You cannot define multiple times the same maintenance kind on an "
+         "equipment maintenance plan!")]
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = 'maintenance.equipment'
+
+    maintenance_plan_ids = fields.One2many(string='Maintenance plan',
+                                           comodel_name='maintenance.plan',
+                                           inverse_name='equipment_id')
+    maintenance_team_required = fields.Boolean(
+        compute='_compute_team_required')
+
+    @api.depends('maintenance_plan_ids')
+    def _compute_team_required(self):
+        for equipment in self:
+            equipment.maintenance_team_required = len(
+                equipment.maintenance_plan_ids) >= 1
+
+    def _create_new_request(self, maintenance_plan):
+        self.ensure_one()
+        self.env['maintenance.request'].create({
+            'name': _('Preventive Maintenance (%s) - %s') %
```

Comment:
Please extract the dict creation into it's own `_prepare_new_maintenance_request` method, so that it is easy to extend it.